### PR TITLE
Fix (4 device) multi-gpu `ShardedGradScaler` Tests in `ciflow/periodic`

### LINKED ISSUE #99427: DISABLED test_fsdp_ddp_parity_with_grad_scaler_offload_false_none_mixed_precision_use_orig_params (__main__.TestShardedGradScalerParityWithDDP)
Platforms: linux

This test was disabled because it is failing on master ([recent examples](http://torch-ci.com/failure/distributed%2Ffsdp%2Ftest_fsdp_sharded_grad_scaler.py%3A%3ATestShardedGradScalerParityWithDDP%3A%3Atest_fsdp_ddp_parity_with_grad_scaler_offload_false_none_mixed_precision_use_orig_params)).

This is failing multigpu test in trunk after https://github.com/pytorch/pytorch/pull/99175.  In the forward fix, please add `Fixes https://github.com/pytorch/pytorch/issues/99427` in your commit message and attach `ciflow/periodic` label to the PR to run the test

cc @mrshenli @pritamdamania87 @zhaojuanmao @satgera @rohan-varma @gqchen @aazzolini @osalpekar @jiayisuse @H-Huang @kwen2501 @awgu

### PR DESCRIPTION
**Pull request by speediedan**
Fixes #99427 

Given the provided CI logs, I ~~suspect~~[^1] `inf` is being hit with the initial (FSDP model) step of the [test in question](https://github.com/pytorch/pytorch/actions/runs/4707887920/jobs/8350225813#step:13:36189). The DDP loss is correct and indicative of two steps being taken but the FSDP loss is approximately half of the loss expected with the first step (suggesting a step was skipped and the scale was halved). I'm further reducing `init_scale` in this PR in order to ~~test the hypothesis~~[^2] (error occurs with 4 device multi-gpu tests only, not the 2 device tests I can verify locally). 

I'll ensure I add the label `ciflow/periodic`[^3] to future PRs I suspect could potentially exhibit divergent behavior with >2 devices. Ideally all tests would be insensitive to device scaling but I recognize for some tests imposing that design constraint might be more trouble than it's worth.

@awgu @huydhn 

[^1]: Suspicion confirmed
[^2]: The relevant periodic tests are [now passing](https://github.com/pytorch/pytorch/actions/runs/4738073998/jobs/8411862508) 
[^3]: Didn't know that existed, great to know!

### CODE REVIEW

**Comment by speediedan** [comment]
As mentioned in the updated description, all relevant `periodic` tests are [now passing](https://github.com/pytorch/pytorch/actions/runs/4738073998/jobs/8411862508). Thanks for catching this and alerting us!  @awgu @huydhn

**Review by huydhn — APPROVED** [review]
LGTM!  Thank you for the fix.  Btw, those errors from other periodic jobs are all unrelated, so it's safe to force merge this once you're ready

**Comment by speediedan** [comment]
> LGTM! Thank you for the fix. Btw, those errors from other periodic jobs are all unrelated, so it's safe to force merge this once you're ready

Thanks @huydhn! I'm not authorized to merge but once someone on the distributed team approves we should be able to move forward. @awgu is out this week so it would need to be someone else if this week, though this probably could wait until next week I would guess.

**Comment by huydhn** [comment]
> Thanks @huydhn! I'm not authorized to merge but once someone on the distributed team approves we should be able to move forward. @awgu is out this week so it would need to be someone else if this week, though this probably could wait until next week I would guess.

No worry, let me see if I can do the merge for you

**Pull request closed without merging**
